src/dbnode/persist/fs/retriever.go

```diff
@@ -470,7 +496,73 @@ func (r *blockRetriever) StreamIndexChecksum(
 	startTime time.Time,
 	nsCtx namespace.Context,
 ) (ident.IndexChecksum, bool, error) {
-	return ident.IndexChecksum{}, false, nil // TODO: implement.
+	req := r.reqPool.Get()
+	req.shard = shard
+	// NB(r): Clone the ID as we're not positive it will stay valid throughout
+	// the lifecycle of the async request.
+	req.id = r.idPool.Clone(id)
+	req.start = startTime
+	req.blockSize = r.blockSize
+
+	req.reqType = streamIdxChecksumReq
+	req.useID = useID
+	req.resultWg.Add(1)
+
+	// Ensure to finalize at the end of request
+	ctx.RegisterFinalizer(req)
+
+	// Capture variable and RLock() because this slice can be modified in the
+	// Open() method
+	r.RLock()
+	// This should never happen unless caller tries to use Stream() before Open()
+	if r.seekerMgr == nil {
+		r.RUnlock()
+		return ident.IndexChecksum{}, false, errNoSeekerMgr
+	}
+	r.RUnlock()
+
+	idExists, err := r.seekerMgr.Test(id, shard, startTime)
+	if err != nil {
+		return ident.IndexChecksum{}, false, err
+	}
+
+	// If the ID is not in the seeker's bloom filter, then it's definitely not on
+	// disk and we can return immediately.
+	if !idExists {
+		return ident.IndexChecksum{}, false, nil
+	}
+
+	reqs, err := r.shardRequests(shard)
+	if err != nil {
+		return ident.IndexChecksum{}, false, err
+	}
+
+	reqs.Lock()
+	reqs.queued = append(reqs.queued, req)
+	reqs.Unlock()
+
+	// Notify fetch loop
+	select {
+	case r.notifyFetch <- struct{}{}:
+	default:
+		// Loop busy, already ready to consume notification
+	}
+
+	// The request may not have completed yet, but it has an internal
+	// waitgroup which the caller will have to wait for before retrieving
+	// the data. This means that even though we're returning nil for error
+	// here, the caller may still encounter an error when they attempt to
+	// read the data.
+	checksum, err := req.waitForIndexChecksum()
```

Comment:
Is this comment still true here? Looks like we're reading the data we want right here (i.e. the checksum) and not later up the call stack like we do with other `Stream` methods.

Reply:
Ish... we do read it here, but it's still [hidden behind a waitgroup](https://github.com/m3db/m3/pull/2632/files/886ab7c36704093bc41d9fd56663e4d1db182e0c#diff-667e8a7e7c5485bf7d69c2f73cbd4887R704) similarly to how we fetch full entries